sway-core/src/language/ty/declaration/trait.rs

```diff
@@ -128,9 +142,15 @@ impl SubstTypes for TyTraitDeclaration {
                 TyTraitInterfaceItem::TraitFn(function_decl_ref) => {
                     let new_decl_ref = function_decl_ref
                         .clone()
-                        .subst_types_and_insert_new(type_mapping, engines);
+                        .subst_types_and_insert_new_with_parent(type_mapping, engines);
                     function_decl_ref.replace_id((&new_decl_ref).into());
                 }
+                TyTraitInterfaceItem::Constant(decl_ref) => {
+                    let new_decl_ref = decl_ref
+                        .clone()
+                        .subst_types_and_insert_new(type_mapping, engines);
```

Comment:
Should this be `subst_types_and_insert_new_with_parent` like above? I'm not sure what the difference is.